src/transfer.js

```diff
@@ -0,0 +1,346 @@
+import { getFileHash } from './utils';
+import axios from 'axios';
+/**
+ * @typedef {Object} BasicLfsAction An LFS basic transfer mode action descriptor
+ * @property {String} href
+ * @property {Object<String, String>} header
+ * @property {Number} expires_in
+ */
+
+class GitLfsBasicTransfer {
+
+  /**
+   * Create a new 'basic' transfer adapter
+   *
+   * @param {Object} actions
+   * @param {BasicLfsAction} actions.upload
+   * @param {BasicLfsAction} actions.verify
+   * @param {File} file
+   * @param {CallableFunction} onProgress
+   */
+  constructor(actions, file, onProgress) {
+    this._file = file;
+    this._onProgress = onProgress;
+
+    this._uploadAction = actions.upload || null;
+    this._verifyAction = actions.verify || null;
```

Comment:
there is no need to use underscore before the variables using javascript because there is no support for encapsulation

Reply:
it just way of saying the variables are private

Reply:
but javascript doesn't have private variables.